ethcore/src/ethereum/ethash.rs

```diff
@@ -322,10 +324,6 @@ impl Engine<EthereumMachine> for Arc<Ethash> {
 			return Err(From::from(BlockError::InvalidProofOfWork(OutOfBounds { min: Some(header.difficulty().clone()), max: None, found: difficulty })));
 		}
 
-		if header.gas_limit() > &0x7fffffffffffffffu64.into() {
```

Comment:
Just a note on this: this would allow miner to craft blocks that have difficulty higher than this limit, but later when this block is importing, it will be rejected. It means that this will be one more place where if a miner configured its client incorrectly, mining will be wasted.

Reply:
I did not think about that, I am restoring this test, it will be redundant with header check, but it is not a costy test.

Reply:
I think this should be fine right now, and I don't think this would be an issue for any live networks. But just wanted to write a note here in case anything wrong happens. :)